node/node_test.go

```diff
@@ -0,0 +1,303 @@
+		AppKey:      testAppKey,
+	}
+	_, err = conf.Validate("")
+	test.That(t, err, test.ShouldBeNil)
+}
+
+func TestValidateABPAttributes(t *testing.T) {
+	// Test missing AppSKey
+	conf := &Config{
+		DecoderPath: testDecoderPath,
+		Interval:    &testInterval,
+		JoinType:    testJoinTypeABP,
+		NwkSKey:     testNwkSKey,
+		DevAddr:     testDevAddr,
+	}
+	_, err := conf.Validate("")
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "app session key is required")
+
+	// Test invalid AppSKey length
+	conf = &Config{
+		DecoderPath: testDecoderPath,
+		Interval:    &testInterval,
+		JoinType:    testJoinTypeABP,
+		AppSKey:     "0123456", // Not 16 bytes
+		NwkSKey:     testNwkSKey,
+		DevAddr:     testDevAddr,
+	}
+	_, err = conf.Validate("")
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "app session key must be 16 bytes")
+
+	// Test missing NwkSKey
+	conf = &Config{
+		DecoderPath: testDecoderPath,
+		Interval:    &testInterval,
+		JoinType:    testJoinTypeABP,
+		AppSKey:     testAppSKey,
+		DevAddr:     testDevAddr,
+	}
+	_, err = conf.Validate("")
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "network session key is required")
+
+	// Test invalid NwkSKey length
+	conf = &Config{
+		DecoderPath: testDecoderPath,
+		Interval:    &testInterval,
+		JoinType:    testJoinTypeABP,
+		AppSKey:     testAppSKey,
+		NwkSKey:     "0123456", // Not 16 bytes
+		DevAddr:     testDevAddr,
+	}
+	_, err = conf.Validate("")
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "network session key must be 16 bytes")
+
+	// Test missing DevAddr
+	conf = &Config{
+		DecoderPath: testDecoderPath,
+		Interval:    &testInterval,
+		JoinType:    testJoinTypeABP,
+		AppSKey:     testAppSKey,
+		NwkSKey:     testNwkSKey,
+	}
+	_, err = conf.Validate("")
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "device address is required")
+
+	// Test invalid DevAddr length
+	conf = &Config{
+		DecoderPath: testDecoderPath,
+		Interval:    &testInterval,
+		JoinType:    testJoinTypeABP,
+		AppSKey:     testAppSKey,
+		NwkSKey:     testNwkSKey,
+		DevAddr:     "0123", // Not 4 bytes
+	}
+	_, err = conf.Validate("")
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "device address must be 4 bytes")
+
+	// Test valid ABP config
+	conf = &Config{
+		DecoderPath: testDecoderPath,
+		Interval:    &testInterval,
+		JoinType:    testJoinTypeABP,
+		AppSKey:     testAppSKey,
+		NwkSKey:     testNwkSKey,
+		DevAddr:     testDevAddr,
+	}
+	_, err = conf.Validate("")
+	test.That(t, err, test.ShouldBeNil)
+}
+
+func TestNewNode(t *testing.T) {
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	// Create mock gateway dependency
+	mockGateway := &inject.Sensor{}
```

Comment:
can we add a GetReadings inject to this to make sure the node can get the correct reading?